autode/opt/coordinates/internals.py

```diff
@@ -0,0 +1,96 @@
+"""
+Internal coordinates. Notation follows
+
+
+x : Cartesian coordinates
+B : Wilson B matrix
+q : Primitive internal coordinates
+"""
+import numpy as np
+from typing import Any
+from abc import ABC, abstractmethod
+from autode.opt.coordinates.base import OptCoordinates
+from autode.opt.coordinates.primitives import InverseDistance
+
+
+class InternalCoordinates(OptCoordinates):
+
+    def __new__(cls, input_array) -> 'InternalCoordinates':
+        """New instance of these internal coordinates"""
+
+        arr = super().__new__(cls, input_array, units=None)
+
+        arr._x = None                # Cartesian coordinates
+        arr.primitive_type = None    # Type of PIC e.g. InverseDistances
+
+        return arr
+
+    def __array_finalize__(self, obj: 'OptCoordinates') -> None:
+        """See https://numpy.org/doc/stable/user/basics.subclassing.html"""
+        self._x = getattr(obj, '_x', None)
+        self.primitive_type = getattr(obj, 'primitive_type', None)
+        return OptCoordinates.__array_finalize__(self, obj)
+
+
+class PIC(list, ABC):
+    """Primitive internal coordinates"""
+
+    def __init__(self,
+                 x: 'autode.opt.CartesianCoordinates'):
+        """
+        Constructor for a set of primitives
+
+        -----------------------------------------------------------------------
+        Arguments:
+            x: Cartesian coordinates
+        """
+        super().__init__()
+
+        self._populate(x)                       # lgtm [py/init-calls-subclass]
```

Comment:
This raises a LGTM alert for possible partial class initalisation, unfortunately I don't see another way to do this. Thoughts?